tests/test_activate.py

```diff
@@ -1,74 +1,194 @@
+            shell_args="/d /c",
+            path_from=path_identity,
+            path_to=path_identity,
+            slash_convert = (u"/", u"\\"),
+        ),
+        "cygwin": dict(unix_shell_base, exe="c:\\cygwin\\bin\\bash", path_from=cygwin_path_to_win, path_to=win_path_to_cygwin),
+        "msys": dict(unix_shell_base, exe="C:\\msys\\1.0\\bin\\bash", path_from=unix_path_to_win, path_to=win_path_to_unix),
+        "msys2": dict(unix_shell_base, exe="C:\\msys\\2.0\\usr\\bin\\bash", path_from=unix_path_to_win, path_to=win_path_to_unix),
+    }
 
-# Only run these tests for commands that are installed.
+else:
+    shells = {
+        "bash": dict(unix_shell_base, exe="bash"),
+        "zsh": dict(unix_shell_base, exe="zsh"),
+    }
+
+def run_in(command, shell):
+    if shell == 'cmd.exe':
+        cmd_script = tempfile.NamedTemporaryFile(suffix='.bat', mode='wt', delete=False)
+        cmd_script.write(command)
+        cmd_script.close()
+        cmd_bits = [shells[shell]["exe"]] + shells[shell]["shell_args"].split(" ") + [cmd_script.name]
+        p = subprocess.Popen(cmd_bits, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = p.communicate()
+        os.unlink(cmd_script.name)
+    elif shell == 'powershell':
+        raise NotImplementedError
+    else:
+        cmd_bits = [shells[shell]["exe"]] + shells[shell]["shell_args"].split(" ") + [translate_stream(command, shells[shell]["path_to"])]
+        print(cmd_bits)
+        p = subprocess.Popen(cmd_bits, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = p.communicate()
+    output_translator = shells[shell]["path_from"]
+    streams = [translate_stream(stream.strip().decode('utf-8').replace('\r\n', '\n').replace('\\\\', '\\'), output_translator)
+                      for stream in (stdout, stderr)]
+    return streams
+
+def assert_equals(a, b, output=""):
+    output = "%r != %r" % (a.lower(), b.lower()) + "\n\n" + output
+    assert a.lower() == b.lower(), output
+
+def assert_not_in(a, b, output=""):
+    assert a.lower() not in b.lower(), "%s %r should not be found in %r" % (output, a.lower(), b.lower())
+
+def assert_in(a, b, output=""):
+    assert a.lower() in b.lower(), "%s %r cannot be found in %r" % (output, a.lower(), b.lower())
+
+
+def gen_test_env_paths(envs, shell, num_test_folders=3):
+    """People need not use all the test folders listed here.
+    This is only for shortening the environment string generation.
+
+    Also encapsulates paths in double quotes.
+    """
+    paths = [join(envs, "test{}".format(test_folder+1)) for test_folder in range(num_test_folders)]
+    for path in paths[:2]:      # Create symlinks ONLY for the first two folders.
+        symlink_conda(path, sys.prefix)
+    converter = shells[shell]["path_to"]
+    paths = [converter(path) for path in paths]
+    return paths
+
+def _envpaths(env_root, env_name=""):
+    if 'win' in platform:
+        paths = [join(env_root, env_name).rstrip("\\"),
```

Comment:
I think it is, because when env_name is "" (the default) - then you end up with the separator at the end.  This gets called a few places with the default:  ```pathsep.join(_envpaths(root_dir))```